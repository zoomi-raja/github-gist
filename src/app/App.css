.container{
  margin: 0 auto;
	max-width: 120rem;
	padding: 2rem;
	color: #777;
}