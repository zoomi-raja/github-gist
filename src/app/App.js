import { useEffect } from 'react';

import './App.css';
import { Header } from './components/header/header';
import { Result } from './components/result/result';
import { getByName, getPublicList } from './utilities/requests';

function App() {
  useEffect(()=>{
    async function fetchData() {
      // You can await here
      const response = await getPublicList()
      const userResp = await getByName({username:'levimm'})
      console.log(response,userResp);
      // ...
    }
    fetchData();
    
  })
  return (
    <div className="App">
      <div className='container'>
        <Header />
        <Result />
      </div>
    </div>
  );
}

export default App;
