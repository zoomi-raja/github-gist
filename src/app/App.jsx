import React from "react";
import { useCallback, useEffect, useRef, useState } from "react";

import "./App.css";
import { Header } from "./components/header/header";
import { Result } from "./components/result/result";
import { debounce } from "./utilities/functions";
import useSearch from "./useSearch";

/** refrence for our local cache remember it will not persist on refresh*/
const cache = new Map();

function App() {
  /**state management we could have used redux but overkill for current scenario useReducer is good substitute
   */
  const [search, setSearch] = useState("");
  const [apiResult, searchResult] = useSearch(cache);
  /** useCallback is used to keep the refrence of method instead of initiating everytime
   */
  const debouncedSearchResult = useCallback(debounce(searchResult, 250), []);

  /**
   * monitor the search input and debounce it
   * this will help to limit the rate of api hits
   */
  useEffect(() => {
    const abortController = new AbortController();
    // get new results if search string length > 3
    debouncedSearchResult(search, abortController.signal);
    // abort on unmount
    return () => {
      abortController.abort();
    };
  }, [search]);
  return (
    <div className="App">
      <div className="container">
        <Header search={search} setSearch={setSearch} />
        <Result {...apiResult} />
      </div>
    </div>
  );
}

export default App;
