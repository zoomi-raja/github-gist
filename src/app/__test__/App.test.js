import renderer from 'react-test-renderer';
import App from '../App';

it('app snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
})