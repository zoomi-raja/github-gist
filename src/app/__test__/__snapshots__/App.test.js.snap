// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <input
        className="header__search"
        onChange={[Function]}
        placeholder="Search By Name..!"
        type="search"
        value=""
      />
    </div>
    <div
      className="results"
    >
      <div
        className="load"
      >
        <div
          className="loader"
        />
      </div>
      <div
        className="no-record"
      >
        <span>
          No Result Found
        </span>
      </div>
    </div>
  </div>
</div>
`;
